core/src/cluster_slot_state_verifier.rs

```diff
@@ -287,28 +357,77 @@ fn check_duplicate_confirmed_hash_against_frozen_hash(
     }
 }
 
+fn check_epoch_slots_hash_against_frozen_hash(
+    state_changes: &mut Vec<ResultingStateChange>,
+    slot: Slot,
+    epoch_slots_frozen_hash: Hash,
+    bank_frozen_hash: Hash,
+    is_dead: bool,
+) {
+    if epoch_slots_frozen_hash != bank_frozen_hash {
+        if is_dead {
```

Comment:
In this case we could still have the correct version right? If we had an incorrect ancestor, and this block was marked dead it is not necessary that this block is duplicate right?

Reply:
yeah, its not necessarily this block that is duplicate, that's correct, this block just gets dumped in the crossfire :'(.

The hope is we eventually make an ancestor request for the earliest dead slot. After we get the ancestors for that earliest dead slot, that should tell us which of the frozen/not dead ancestors is bad.